jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java

```diff
@@ -833,6 +834,24 @@ public void testDump() throws Exception
         assertThat(count(dump, "QueuedThreadPoolTest.lambda$testDump$"), is(1));
     }
 
+    @Test
+    public void testContextClassLoader() throws Exception
+    {
+        QueuedThreadPool tp = new QueuedThreadPool();
+        tp.setMinThreads(1);
+        tp.setMaxThreads(2);
+        tp.setIdleTimeout(1000);
+        tp.setThreadsPriority(Thread.NORM_PRIORITY - 1);
+        try (StacklessLogging stackless = new StacklessLogging(QueuedThreadPool.class))
+        {
+            tp.start();
+            tp.execute(() ->
+            {
+                assertThat(Thread.currentThread().getContextClassLoader(), Matchers.equalTo(QueuedThreadPool.class.getClassLoader()));
+            });
+        }
+    }
```

Comment:
This test would have passed also before the change, right?
The real test should really create a different classloader, use a pooled thread, change its classloader, spawn _another_ thread and verify the spawned thread does not inherit the context classloader.

Reply:
Made the test even simpler.